src/plugins/auth/auth.ts

```diff
@@ -262,6 +316,32 @@ const plugin: FastifyPluginAsync<AuthPluginOptions> = async (fastify, options) =
       },
     );
 
+    // loginpassword
+    fastify.post<{ Body: { email: string; password: string } }>(
+      '/loginpassword',
+      { schema: passswordLogin },
+      async ({ body, log }, reply) => {
+        const email = body.email.toLowerCase();
+        const task = memberTaskManager.createGetByTask(GRAASP_ACTOR, { email });
+        const [member] = await runner.runSingle(task, log);
+
+        if (member) {
+          const link = await verifyCredentials(member, body, null);
+          if (link === null) {
+            reply.status(StatusCodes.NOT_ACCEPTABLE).send(ReasonPhrases.NOT_ACCEPTABLE);
+          } else if (link) {
+            reply.status(StatusCodes.SEE_OTHER).send({ link });
+          } else {
+            reply.status(StatusCodes.UNAUTHORIZED).send(ReasonPhrases.UNAUTHORIZED);
+          }
+        } else {
+          const { email } = body;
+          log.warn(`Login attempt with non-existent email '${email}'`);
+          reply.status(StatusCodes.NOT_FOUND).send(ReasonPhrases.NOT_FOUND);
```

Comment:
I prefer returning the result as soon as possible, so we avoid encapsulating ifs, especially if you mis errors and success cases. Could you refactor it and take advantage of `return reply...`?